services/self-service/src/main/java/com/epam/dlab/backendapi/dao/OdahuDAOImpl.java

```diff
@@ -77,15 +80,20 @@
 
     @Override
     public OdahuFieldsDTO getFields(String name, String project, String endpoint) {
-        Optional<Document> one = findOne(PROJECTS_COLLECTION, odahuProjectEndpointCondition(name, project, endpoint),
-                fields(include(ODAHU_FIELD), excludeId()));
-        OdahuFieldsDTO odahuFields = null;
-        if (one.isPresent()) {
-            List<OdahuFieldsDTO> list = convertFromDocument(one.get().get(ODAHU_FIELD, ArrayList.class), new TypeReference<List<OdahuFieldsDTO>>() {
-            });
-            odahuFields = list.get(0);
-        }
-        return odahuFields;
+        FindIterable<Document> odahuDoc = find(PROJECTS_COLLECTION,
+                odahuProjectEndpointCondition(name, project, endpoint),
+                fields(include(ODAHU_FIELD), excludeId())); //TODO: Potentially, method can be improved by filtering the Documents
```

Comment:
Could you please implement a method **find(String collection, Bson condition, Bson projection, Class<T> resultedClass)** in BaseDAO class?

Reply:
In context of this issue such interface is not applicable. Cause document conversion interface requires the TypeReference object.
Plus, the ODAHU clusters array list field is extracted from the Document:
odahuClusters.get(ODAHU_FIELD, ArrayList.class)
In frame of this ticket, we can leave legacy methods